src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -252,7 +252,8 @@ internal static bool IsRecoverableException(Exception exception)
         {
             return exception is SecurityTokenInvalidSignatureException
                   || exception is SecurityTokenInvalidIssuerException
-                  || exception is SecurityTokenSignatureKeyNotFoundException;
+                  || exception is SecurityTokenSignatureKeyNotFoundException
+                  || exception is SecurityTokenDecryptionFailedException;
```

Comment:
I don't think that this is sufficient. We should signal metadata refresh only if encryption keys are part of metadata.